test/e2e/util/kubectl.go

```diff
@@ -62,16 +62,28 @@ func KubectlDelete(kubeconfigPath, namespace string, args []string) error {
 	return err
 }
 
-// KubectlExec executes "kubectl exec" given a list of arguments.
-func KubectlExec(kubeconfigPath, podName, namespace string, args []string) (string, error) {
-	args = append([]string{
+func KubectlExecCurl(kubeconfigPath, podName string, endpoint string) (string, error) {
```

Comment:
Accept namespace as an argument here. You'll use `http://otel-collector:8889/metrics` as the endpoint, which works as long as the curl pod and the otel-collector are in the same namespace.